.github/workflows/ci-docker.yml

```diff
@@ -50,6 +50,17 @@ jobs:
           username: ${{ inputs.dockerhub-username }}
           password: ${{ secrets.CI_DOCKERHUB_TOKEN }}
 
+      - name: Go Build Cache for Docker
+        uses: actions/cache@v3
+        with:
+          path: go-build-cache
+          key: ${{ runner.os }}-go-build-cache-${{ hashFiles('**/go.sum') }}
+
+      - name: inject go-build-cache into docker
+        uses: reproducible-containers/buildkit-cache-dance@v2.1.2
+        with:
+          cache-source: go-build-cache
```

Comment:
Do the Dockerfiles need to be updated for this to work?

With those `--mount=type=cache,target=...` [flags for RUN commands](https://docs.docker.com/build/ci/github-actions/cache/#cache-mounts)

Reply:
they have the appropriate cache mounts, which are currently working and used when building locally. this addition (hopefully) gives the github action caching powers as well

Reply:
ahh yes I missed it, mistakenly only checked the rocksdb-base 😄 seems good then!

Reply:
yee! i think some sort of cache mount for the building-rocksdb-command is what would allow caching of that container layer, but i didn't take the time to figure out what 😂 just opted to extract as a separate container for the same result..